File: Java/ToyLanguageGUI_exam_final/src/domain/statements/AssignStmt.java
package domain.statements;

import domain.expressions.Expression;
import domain.PrgState;
import utils.IDictionary;
import utils.IHeap;

public class AssignStmt implements Statement {
    private String varName;
    private Expression expr;

    public AssignStmt(String v,Expression e)
    {
        this.varName = v;
        this.expr = e;
    }

    @Override
    public PrgState execute(PrgState p){
        //try {
            IDictionary<String, Integer> symb = p.getSymbols();
            IHeap<Integer,Integer> h = p.getHeap();
            int result = expr.eval(symb,h);
            if (symb.contains(varName))
                symb.update(varName, result);
            else
                symb.add(varName, result);
        //}catch (StackException se){
        //System.out.println(se.getMessage());}
        return null;
    }

    @Override
    public String toString(){
        return varName + '=' +expr;
    }
}

File: ToyInterpretor/ToyInterpretor/Domain/Statements/AssignStmt.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ToyInterpretor.Domain.Expressions;
using ToyInterpretor.Utils;

namespace ToyInterpretor.Domain.Statements
{
    class AssignStmt : Statement
    {
        private String varName;
        private Expression expr;

        public AssignStmt(String v, Expression e)
        {
            this.varName = v;
            this.expr = e;
        }

        public PrgState execute(PrgState p)
        {
            ISymTable<String, int> symb = p.getSymbols();
            int result = expr.eval(symb);
            if (symb.contains(varName))
                symb.update(varName, result);
            else
                symb.add(varName, result);
           
            return p;
        }

      
        public  override String ToString()
        {
            return varName + '=' + expr;
        }
    }
}
